### PR TITLE
[FIX] mrp,mrp_workorder: Consumed SN shouldn't appear in the dropdown

### PR DESCRIPTION
**Pull request by macadelic80**
What are the steps to reproduce your issue ?

    1. Install "mrp_workorder"
    2. Create new product "sn" with traceability to "By Unique Serial Number"
    3. Create new MO with new bill of material with more than one quantity to produce
    4. Process the workorder

What is currently happening ?

    The list of SNs does not depend on the SNs already in use

What are you expecting to happen ?

    The system shows only SN that can be used

Why is this happening ?

    The system displays all the sn corresponding to the product

How to fix the bug ?

    Show only unused SNs in current production

Linked PR (CE): https://github.com/odoo/enterprise/pull/15174

opw-2410301

### CODE REVIEW

**Comment by abz-odoo** [comment]
fw-bot ignore

File: addons/mrp/models/mrp_workorder.py [hunk 1]
```diff
@@ -178,6 +179,24 @@ def _compute_scrap_move_count(self):
...
+        Restrict the selectable lot to the lot/sn used in other workorders.
+        """
+        productions = self.mapped('production_id')
+        treated = self.browse()
+
+        for production in productions:
+            if production.product_id.tracking == 'none':
+                continue
+            allowed_lot_ids = self.env['stock.production.lot'].search([('product_id', '=', production.product_id.id)])
+            used = productions.mapped("move_finished_ids").mapped("move_line_ids").mapped('lot_id')
```
Comment (Whenrow) [inline addons/mrp/models/mrp_workorder.py:193]: filter the move_line_ids with qty_done > 0 to be sure

**Review by Whenrow — APPROVED** [review]

**Comment by abz-odoo** [comment]
@nim-odoo @nle-odoo

**Review by nle-odoo — COMMENTED** [review]
testing it, I still see already used serial numbers:

> 2020-12-08-120927_1920x1080_scrot

> 2020-12-08-121210_830x667_scrot

and the final_lot_id in the workorder normal form view already has a domain that I think has priority:

https://github.com/odoo/odoo/blob/7ddf36474596ec5213aba7b892c1a62998c99da9/addons/mrp/views/mrp_workorder_views.xml#L145

maybe I'm missing something to reproduce or understand the issue.

__edit:__

as we discussed, I had not noticed seen the enterprise PR: https://github.com/odoo/enterprise/pull/15174

**Comment by nle-odoo** [comment]
that didn't work because I did not have the enterprise PR

then `<field name="allowed_lots_domain" invisible="1"/>` doesn't seem to be used in mrp.mrp_production_workorder_form_view_inherit (since there is also a custom domain)

changing the domain on the python field also doesn't seem to change much (and possibly could break customization were final_lot_id appear without allowed_lots_domain?)

in enterprise, the `use_next_on_work_order_id` feature is broken by this since the domain is removed:

```diff
- ['|', '&amp;', ('product_id', '=', product_id), '|', ('product_id.tracking', '!=', 'serial'), '&amp;', ('use_next_on_work_order_id', '=', False), ('use_next_on_work_order_id', '!=', 'is_first_wo'), ('use_next_on_work_order_id', '=', id)]
+ [('id', 'in', allowed_lots_domain)]
```

with your change

maybe it should still be working and you should just replace `('product_id', '=', product_id)` by `('id', 'in', allowed_lots_domain)`

**Pull request closed without merging**